src/libaktualizr/bootloader/bootloader.cc

```diff
@@ -10,7 +10,7 @@
 #include "utilities/exceptions.h"
 #include "utilities/utils.h"
 
-Bootloader::Bootloader(const BootloaderConfig& config, INvStorage& storage) : config_(config), storage_(storage) {
+Bootloader::Bootloader(BootloaderConfig config, INvStorage& storage) : config_(std::move(config)), storage_(storage) {
```

Comment:
kind of a c++ learning question for my benefit: Seems like passing as a const by reference would be more ideal. What benefit does this gain?

Reply:
I guess it comes from the clang-tidy best practice, the reason is when you pass value through const references the move semantic won't have an effect and copy constructor will be called anyway.
This can have an effect when you pass rvalue into constructor, in this case this approach avoids calling the copy constructor.

Reply:
Yes, exactly, it comes from [this clang-tidy warning](https://clang.llvm.org/extra/clang-tidy/checks/modernize-pass-by-value.html).
Which it turn was caused by the change in src/libaktualizr/bootloader/bootloader.h:28
Previously, we stored a reference to the config object, which isn't the right thing to do, IMO.  It makes the behavior of Bootloader class dependent on the state of an external object and that is usually not what we want.

Reply:
thanks that was exactly what i was hoping to learn about.